### PR TITLE
Add `ignore_words` options to L057 and L059

### PR DESCRIPTION
**Pull request by tunetheweb**
### Brief summary of the change made

Makes progress on #2747

### Are there any other side effects of this change that we should be aware of?
Nope

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/rules/L057.py [hunk 1]
```diff
@@ -57,16 +58,29 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
...
+            # very slow.
+            ignore_words_list = self._init_ignore_words_list()
+
         # Assume unquoted (we'll update if quoted)
         policy = self.unquoted_identifiers_policy
 
         identifier = context.segment.raw
 
+        # Skip if in ignore list
+        if ignore_words_list and identifier in ignore_words_list:
```
Comment (barrywhart) [inline src/sqlfluff/rules/L057.py:81]: Is this intended to be case insensitive? I notice we are lowercasing the strings in the list, but apparently not `identifier`. Seems like we should do both or neither.

File: src/sqlfluff/rules/L059.py [hunk 2]
```diff
@@ -70,28 +70,41 @@ class Rule_L059(BaseRule):
...
+        # Get the ignore_words_list configuration.
+        try:
+            ignore_words_list = self.ignore_words_list
+        except AttributeError:
+            # First-time only, read the settings from configuration. This is
+            # very slow.
+            ignore_words_list = self._init_ignore_words_list()
+
+        # Skip if in ignore list
+        if ignore_words_list and identifier_contents in ignore_words_list:
```
Comment (barrywhart) [inline src/sqlfluff/rules/L059.py:105]: Same question

File: test/fixtures/rules/std_rule_cases/L057.yml [hunk 3]
```diff
@@ -320,3 +320,32 @@ test_pass_alteration_identifier_in_file_reference_spark3:
   configs:
     core:
       dialect: spark3
+
+test_pass_ignore_lists_quoted:
```
Comment (barrywhart) [inline test/fixtures/rules/std_rule_cases/L057.yml:324]: Re: My earlier comment, can we add tests for upper and/or mixed-case strings?

File: test/fixtures/rules/std_rule_cases/L059.yml [hunk 4]
```diff
@@ -261,3 +261,11 @@ test_fail_create_quoted_user_exasol:
   configs:
     core:
       dialect: exasol
+
+test_pass_ignore_lists:
```
Comment (barrywhart) [inline test/fixtures/rules/std_rule_cases/L059.yml:265]: Ditto

**Review by barrywhart — CHANGES_REQUESTED** [review]

File: src/sqlfluff/rules/L057.py (hunk 1 above)
Comment (tunetheweb) [reply src/sqlfluff/rules/L057.py:81]: Good shout. Will add.

Comment (tunetheweb) [reply src/sqlfluff/rules/L057.py]: Added

File: test/fixtures/rules/std_rule_cases/L057.yml (hunk 3 above)
Comment (tunetheweb) [reply test/fixtures/rules/std_rule_cases/L057.yml:324]: Fixed

File: src/sqlfluff/rules/L059.py (hunk 2 above)
Comment (tunetheweb) [reply src/sqlfluff/rules/L059.py]: Fixed

File: test/fixtures/rules/std_rule_cases/L059.yml (hunk 4 above)
Comment (tunetheweb) [reply test/fixtures/rules/std_rule_cases/L059.yml:265]: Fixed.

**Review by barrywhart — APPROVED** [review]

**Pull request merged**